aws/data_source_aws_lambda_invoke.go

```diff
@@ -0,0 +1,91 @@
+package aws
+
+import (
+	"crypto/md5"
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsLambdaInvoke() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsLambdaInvokeRead,
+
+		Schema: map[string]*schema.Schema{
+			"function_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"qualifier": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "$LATEST",
+			},
+
+			"input": {
+				Type:     schema.TypeMap,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"result": {
```

Comment:
Same here -- we should convert this to `TypeString` and remove the JSON unmarshal handling/errors from the data source.

Reply:
If we use `TypeString` for `result`, how the returned data will be used later in terraform code? As far as I know there is no `jsondecode` or similar function available.